core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -92,6 +92,9 @@ private void onWrapperCompleted(HttpResponseWrapper resWrapper, int id, @Nullabl
         resWrapper.onSubscriptionCancelled(cause);
 
         if (cause != null) {
+            if (cause instanceof UnprocessedRequestException) {
```

Comment:
We may not need to send RST_STREAM when the stream was closed or a RST_STREAM was received before. I was wondering if our logic already handles the cases. 
https://github.com/line/armeria/blob/e08527d480482f345007ad5d8c68062ae8f5b3f4/core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java#L138-L150
https://github.com/line/armeria/blob/e08527d480482f345007ad5d8c68062ae8f5b3f4/core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java#L314

Reply:
I think Netty handles those cases:
https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java#L808-L810

Reply:
Netty seems to send another RST_STREAM by `encoder.writeRstStream()` when the client receives an RST_STREAM or the connection is closed.

My suggestion is:
```java
if (cause instanceof UnprocessedRequestException || 
    cause instanceof ClosedStreamException) {
   return;
}
```

Reply:
Thanks addressed. 👍